### PR TITLE
Added python wrapper for SM script

### PR DESCRIPTION
**Pull request by piyushghai**
Added wrapper for SageMaker. 

```
SM_DATA_ROOT = '/opt/ml/input/data/train'

hyperparameters={
    "sagemaker_distributed_dataparallel_enabled": True,
    "mode": "train" ,
    "checkpoint": '/'.join([SM_DATA_ROOT, 'model/resnet/resnet-nhwc-2018-02-07/model.ckpt-112603']),
    "eval_samples": 5000,
    "init_learning_rate": 0.04,
    "learning_rate_steps": "3750,5000",
    "model_dir_ckpt": "/opt/ml/code/checkpoints/tensorflow_mask_rcnn",
    "num_steps_per_eval": 462,
    "total_steps": 50,
    "train_batch_size": 4,
    "eval_batch_size": 8,
    "training_file_pattern": '/'.join([SM_DATA_ROOT, "train"]),
    "validation_file_pattern": '/'.join([SM_DATA_ROOT, "val"]),
    "val_json_file": '/'.join([SM_DATA_ROOT, 'annotations/instances_val2017.json']),
    "seed": 987
}


estimator = TensorFlow(entry_point='DeepLearningExamples/TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py',
                        role=role,
                        image_uri=docker_image,
                        source_dir='.',
                        framework_version='2.3.1',
                        train_instance_count=2,
                        train_instance_type=instance_type,
                        sagemaker_session=sagemaker_session,
                        subnets=subnets,
                        hyperparameters=hyperparameters,
                        security_group_ids=security_group_ids)
```

### CODE REVIEW

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        try:
+            v = int(v0)
+            values.append(v)
+        except ValueError as err:
+            raise ArgumentError('Invalid value %s, values must be a number' % v)
+    return values
+
+def main():
+    # CLI flags
+    print ("Here")
```
Comment (karan6181) [inline TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:35]: nit: please remove it if not needed.

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    parser = argparse.ArgumentParser(description="MaskRCNN train benchmark")
+    parser.add_argument('--mode', type=str, default='train')
```
Comment (karan6181) [inline TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:37]: Should we add the description from `mask_rcnn_main.py` file here ?

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        f' {"--amp" if flags.amp else ""}'
+        f' {"--use_batched_nms" if flags.use_batched_nms else ""}'
+        f' {"--xla" if flags.xla else ""}'
+        f' {"--nouse_custom_box_proposals_op" if flags.nouse_custom_box_proposals_op else ""}'
+        f' --seed={flags.seed}'
+    )
+
+    # print command
+    line = '-' * shutil.get_terminal_size()[0]
+    print(line, cmd, line, sep='\n')
```
Comment (karan6181) [inline TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:88]: Is this for debugging purpose ?

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py (hunk 1 above)
Comment (piyushghai) [reply TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py]: Done

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py (hunk 2 above)
Comment (piyushghai) [reply TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:36]: We can add those. I'll do it in a separate PR. 
For now, let's get this to run on SM :)

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+def csv_int(vstr, sep=','):
+    values = []
+    for v0 in vstr.split(sep):
+        try:
+            v = int(v0)
+            values.append(v)
+        except ValueError as err:
+            raise ArgumentError('Invalid value %s, values must be a number' % v)
+    return values
+
```
Comment (sandeep-krishnamurthy) [inline TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:32]: Can you please 1 or 2 line of description to say what was the problem and why this parsing was required. Makes it more explicit for other users and team members.

**Review by sandeep-krishnamurthy — APPROVED** [review]
Thank you _/\_

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py (hunk 4 above)
Comment (piyushghai) [reply TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:35]: Done. e5dfc63c93165bf23a058c3fc5b34b7a2928bd4a

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py (hunk 2 above)
Comment (karan6181) [reply TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_sm.py:39]: Sure

**Review by karan6181 — APPROVED** [review]

**Review by rondogency — APPROVED** [review]

**Pull request merged**